File: Atcoder_#330/ruby/b_minimize_abs_1.rb
n, l, r = gets.split(" ").map(&:to_i)
a = gets.split.map(&:to_i)

ans = []
n.times do |i|
  if a[i] < l
    ans << l
  elsif r < a[i]
    ans << r
  else
    ans << a[i]
  end
end

puts ans.join(" ")

File: Atcoder_#330/javascript/b_minimize_abs_1.js
const { endianness } = require('os');

function main(input) {
  let lines = input.split('\n');
  let [n, l, r] = lines[0].trim().split(' ').map(Number);
  let a = lines[1].trim().split(' ').map(Number);

  let ans = [];

  for (let i = 0; i < n; i++) {
    if (a[i] < l) {
      ans.push(l)
    } else if (a[i] > r) {
      ans.push(r)
    } else (
      ans.push(a[i])
    )
  }

  console.log(ans.join(' '));
}

main(require('fs').readFileSync('/dev/stdin', 'utf8'));
